### ISSUE TITLE
api error: Failed to subscribe to partition

### ISSUE THREAD

**Issue by abunjevac** [liftbridge-io/liftbridge#323]
Subscribe() to a stream fails with the following error:

`Failed to lookup offset for timestamp: failed to find log segment for timestamp: EOF`

This happens only if I try to subscribe with StartAtTime(). However, subscribe with StartAtOffset() works OK.
The issue arises after a few days of the stream inactivity (no incoming events) and the subscriber is restarted.

This is a 3 node cluster running on kubernetes.
Image: `liftbridge/liftbridge-k8s:v1.4.1`

However, I did experience the same issue with a local docker container running a single node server based on the latest liftbridge/standalone-dev image (v.1.5.0).

These are kubernetes deployment details.

Config is pretty basic:
```
listen: 0.0.0.0:9292
logging.level: info

nats.servers:
  - "nats://user:secret@nats-client:4222"

clustering.min.insync.replicas: 2
```

The error is logged in liftbridge-1 log:

`time="2021-01-26 15:34:37" level=error msg="api: Failed to subscribe to partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]: rpc error: code = Internal desc = Failed to lookup offset for timestamp: failed to find log segment for timestamp: EOF"`

liftbridge-0 log:
```
time="2021-01-19 05:59:31" level=warning msg="fsm: Removed replica cluster-liftbridge-0 from ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:31" level=error msg="ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] has shrunk below minimum size 2, currently 1"
time="2021-01-19 05:59:31" level=warning msg="fsm: Removed replica cluster-liftbridge-2 from ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:46" level=info msg="ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] has recovered from being below minimum size 2, currently 2"
time="2021-01-19 05:59:46" level=info msg="fsm: Added replica cluster-liftbridge-0 to ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:46" level=info msg="fsm: Added replica cluster-liftbridge-2 to ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
```

liftbridge-1 log:
```
time="2021-01-19 05:59:08" level=error msg="Replica cluster-liftbridge-0 for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] exceeded max lag time (last seen: 15.000115667s, last caught up: 15.000115667s), removing from ISR"
time="2021-01-19 05:59:18" level=error msg="Replica cluster-liftbridge-2 for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] exceeded max lag time (last seen: 15.000100705s, last caught up: 15.000100705s), removing from ISR"
time="2021-01-19 05:59:31" level=warning msg="fsm: Removed replica cluster-liftbridge-0 from ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:31" level=error msg="ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] has shrunk below minimum size 2, currently 1"
time="2021-01-19 05:59:31" level=warning msg="fsm: Removed replica cluster-liftbridge-2 from ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:33" level=error msg="Unable to commit messages for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0], ISR size (1) below minimum (2)"
time="2021-01-19 05:59:46" level=info msg="Replica cluster-liftbridge-0 for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] caught back up with leader, rejoining ISR"
time="2021-01-19 05:59:46" level=info msg="Replica cluster-liftbridge-2 for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] caught back up with leader, rejoining ISR"
time="2021-01-19 05:59:46" level=info msg="ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] has recovered from being below minimum size 2, currently 2"
time="2021-01-19 05:59:46" level=info msg="fsm: Added replica cluster-liftbridge-0 to ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:46" level=info msg="fsm: Added replica cluster-liftbridge-2 to ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-26 15:34:37" level=error msg="api: Failed to subscribe to partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]: rpc error: code = Internal desc = Failed to lookup offset for timestamp: failed to find log segment for timestamp: EOF"
```

liftbridge-2 log:
```
time="2021-01-19 05:59:31" level=warning msg="fsm: Removed replica cluster-liftbridge-0 from ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:31" level=error msg="ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] has shrunk below minimum size 2, currently 1"
time="2021-01-19 05:59:31" level=warning msg="fsm: Removed replica cluster-liftbridge-2 from ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:46" level=info msg="ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0] has recovered from being below minimum size 2, currently 2"
time="2021-01-19 05:59:46" level=info msg="fsm: Added replica cluster-liftbridge-0 to ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
time="2021-01-19 05:59:46" level=info msg="fsm: Added replica cluster-liftbridge-2 to ISR for partition [subject=ems.helb.sensors, stream=ems.helb.sensors, partition=0]"
```

**Comment by tylertreat** [comment]
Thanks for the detailed issue. Will take a look.

**Comment by abunjevac** [comment]
Hi!
Have you managed to look into this? I am still experiencing this. Every now and then I have to publish a dummy record to a stream so my consumers can subscribe with a specific timestamp.

**Comment by tylertreat** [comment]
Could you provide a bit more information on the issue? Namely, is the client subscribing with the same time using `StartAtTime` each time it is restarted? How many days of inactivity before the issue occurs? What do the contents of the partition in question look like, both before the issue occurs and after? If you have client code that can reproduce the issue that would be helpful as well.

Also, you may want to enable debug logging to see if anything else is revealed that could indicate what is happening.

**Comment by tylertreat** [comment]
Got it, thanks. Managed to reproduce the issue. Now I should be able to figure out what's going on and get a fix made. I'll let you know what I find momentarily.

**Comment by tylertreat** [comment]
This should be fixed with #349 which is now in the [latest standalone-dev image](https://hub.docker.com/layers/liftbridge/standalone-dev/latest/images/sha256-378e09bc1f02ddc1d774e3309fcab522a9902bc526011bff114e3d1d494e95b7?context=repo). If you still encounter the problem, please reopen this issue.

**Comment by abunjevac** [comment]
Works as expected, thanks.
